01-JS-DOM-APIs/javascript.js

```diff
@@ -0,0 +1,117 @@
+//1-4
+function show(){
+                  let obj = document.getElementById("hidden");
+                  if(obj.style.display == "block"){
+                        obj.style.display = "none";
+                        document.getElementById("buttonShow").value= "Show";
+                      }
+                  else{
+                        obj.style.display = "block";
+                        document.getElementById("buttonShow").value= "Hide";
+                      }
+                    }
+//5-6
+function buttonJokeWithoutPromise() {
+    let xhttp = new XMLHttpRequest();
+    xhttp.open("GET", "http://api.icndb.com/jokes/random", true);
+    xhttp.onreadystatechange = function(event) {
+    if (xhttp.readyState === XMLHttpRequest.DONE)
+    {
+      let response = JSON.parse(event.target.response);
+      let txt = response.value.joke;
+      document.getElementById("jokeHidden").innerHTML = response.value.joke;
+      document.getElementById("buttonjoke").value="Another joke";
+               }
+           }
+    xhttp.send();
+      }
+//7-9
+function buttonJokeWithPromise(){
+            let config = {
+              url: "http://api.icndb.com/jokes/random"
+                  };
+                        let xhttp = new XMLHttpRequest();
+                        function resolve(xhttp) {
+                            let response = JSON.parse(event.target.response);
+                             document.getElementById("jokeHidden2").innerHTML = response.value.joke;
+                             if (xhttp.readyState == XMLHttpRequest.DONE) {
+                               if (xhttp.responseText.value) {
+                                 document.getElementById("button2").innerHTML  = xhttp.responseText.value.joke;
+                               }
+                             }
+                         }
+                         function reject() {
+                           document.getElementById("all").style.backgroundColor = "red";
```

Comment:
same as above, you can have your red background in a class, and add that class to the *all* node